corehq/util/tests/test_couch_helpers.py

```diff
@@ -0,0 +1,117 @@
+from django.test import TestCase
+
+from couchdbkit import ResourceConflict, ResourceNotFound
+from corehq.util.couch_helpers import ResumableDocsByTypeIterator, TooManyRetries
+from dimagi.utils.couch.database import get_db
+
+
+class TestResumableDocsByTypeIterator(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.db = get_db()
+        cls.docs = []
+        for i in range(3):
+            cls.create_doc("Foo", i)
+            cls.create_doc("Bar", i)
+            cls.create_doc("Baz", i)
+        cls.doc_types = ["Foo", "Bar", "Baz"]
+
+    @classmethod
+    def tearDownClass(cls):
+        for doc_id in set(d["_id"] for d in cls.docs):
+            try:
+                cls.db.delete_doc(doc_id)
+            except ResourceNotFound:
+                pass
+
+    def setUp(self):
+        self.domain = "TEST"
+        self.sorted_keys = ["{}-{}".format(n, i)
```

Comment:
note to self: `self.domain` is not used. `self.sorted_keys = ...` could be moved into `setUpClass`